vision/beta/cloud-client/src/main/java/com/example/vision/AsyncBatchAnnotateImagesGcs.java

```diff
@@ -0,0 +1,161 @@
+
+      ImageSource imgSource = ImageSource.newBuilder().setImageUri(gcsSourcePath).build();
+
+      Image image = Image.newBuilder().setSource(imgSource).build();
+
+      // Set the GCS destination path for where to save the results.
+      GcsDestination gcsDestination =
+          GcsDestination.newBuilder().setUri(gcsDestinationPath).build();
+
+      // Create the configuration for the output with the batch size.
+      // The batch size sets how many pages should be grouped into each json output file.
+      OutputConfig outputConfig =
+          OutputConfig.newBuilder().setGcsDestination(gcsDestination).setBatchSize(2).build();
+
+      // Select the Features required by the vision API
+      Feature features =
+          Feature.newBuilder()
+              .setType(Type.LABEL_DETECTION)
+              .setType(Type.TEXT_DETECTION)
+              .setType(Type.IMAGE_PROPERTIES)
+              .build();
+
+      // Build the request
+      AnnotateImageRequest annotateImageRequest =
+          AnnotateImageRequest.newBuilder().setImage(image).addFeatures(features).build();
+
+      requests.add(annotateImageRequest);
+      AsyncBatchAnnotateImagesRequest request =
+          AsyncBatchAnnotateImagesRequest.newBuilder()
+              .addAllRequests(requests)
+              .setOutputConfig(outputConfig)
+              .build();
+
+      ApiFuture<Operation> future = client.asyncBatchAnnotateImagesCallable().futureCall(request);
+      // Wait for the request to finish. (The result is not used, since the API saves the result to
+      // the specified location on GCS.)
+      Operation response = future.get(180, TimeUnit.SECONDS);
+
+      System.out.println("Waiting for the operation to finish.");
+
+      // Once the request has completed and the output has been
+      // written to GCS, we can list all the output files.
+      Storage storage = StorageOptions.getDefaultInstance().getService();
+
+      // Get the destination location from the gcsDestinationPath
+      Pattern pattern = Pattern.compile("gs://([^/]+)/(.+)");
+      Matcher matcher = pattern.matcher(gcsDestinationPath);
+
+      if (matcher.find()) {
+        String bucketName = matcher.group(1);
+        String prefix = matcher.group(2);
+
+        // Get the list of objects with the given prefix from the GCS bucket
+        Bucket bucket = storage.get(bucketName);
+        Page<Blob> pageList = bucket.list(BlobListOption.prefix(prefix));
+
+        Blob firstOutputFile = null;
+
+        // List objects with the given prefix.
+        System.out.println("Output files:");
+        for (Blob blob : pageList.iterateAll()) {
+          System.out.println(blob.getName());
+
+          // Process the first output file from GCS.
+          // Since we specified batch size = 2, the first response contains
+          // the first two image requests
+          if (firstOutputFile == null) {
+            firstOutputFile = blob;
```

Comment:
Is `firstOutputFile` needed or can just `blob` be used?

Reply:
Would need the firstOutputFile since we are iterating through a list and currently there isn't a method to just get one element.